src/frontend/apps/lti_site/components/TeacherLiveRecordingActions/StopRecording/index.tsx

```diff
@@ -47,12 +47,15 @@ export const StopRecording = () => {
   const [segmentDuration, setSegmentDuration] = useState(0);
   const [recordingActionEnabled, setRecordingActionEnabled] = useState(false);
   const video = useCurrentVideo();
-  useVideoMetadata(intl.locale, {
+  const videoMetaData = useVideoMetadata(intl.locale, {
     onSuccess: (videoMetadata) => {
       setSegmentDuration(videoMetadata.live.segment_duration_seconds);
     },
   });
+
   useEffect(() => {
+    videoMetaData.refetch();
```

Comment:
I disagree with this solution. The API should not be fetch again, this value is not going to change. If the issue is in this `useEffect` you should look for a solution how to fix it. You have all the data you need, you know this component is re render, you should think about how to change the useEffect knowing all of this.

The issue is, IMO, in the `onSuccess` used on the `useVideoMetadata` react query hook. We have an agressive cache on this query, all refetch is disabled and the cache is infinite. IMO only the data returned by `useVideoMetadata` should be used and stop relying on the `segmentDuration` state. 

```
const { data } = useVideoMetadata(intl.locale);

useEffect(() => {
    if (!data?.live.segment_duration_seconds) {
      return;
    }

    let timeoutId: number;
    if (data.live.segment_duration_seconds > 0) {
      timeoutId = window.setTimeout(() => {
        setRecordingActionEnabled(true);
      }, data.live.segment_duration_seconds * 1000);
    }

    return () => {
      window.clearTimeout(timeoutId);
    };
  }, [data?.live.segment_duration_seconds, recordingActionEnabled]);
```

Something like that, it's just an idea and you should investigate more with this.

Reply:
You were right ! it was the problem, didn't think of that.
I fixed it the way you said and it works well :+1: